src/xma/src/xmaapi/xma_utils.cpp

```diff
@@ -108,27 +126,43 @@ namespace xma_core {
         return XMA_ERROR;
     }
 
+    /// <summary>
+    /// create_session_execbo() : Creates XmaHwExecBO object with xrt::Kernel.
+    /// </summary>
+    /// <param name="priv"></param>
+    /// <param name="count"></param>
+    /// <param name="prefix"></param>
+    /// <returns></returns>
     int32_t create_session_execbo(XmaHwSessionPrivate *priv, int32_t count, const std::string& prefix) {
-        for (int32_t d = 0; d < count; d++) {
-            xclBufferHandle  bo_handle = 0;
-            int       execBO_size = MAX_EXECBO_BUFF_SIZE;
-            char     *bo_data;
-            bo_handle = priv->dev_handle.get_handle()->alloc_bo(execBO_size, XCL_BO_FLAGS_EXECBUF);
-            if (!bo_handle || bo_handle == NULLBO)
-            {
-                xma_logmsg(XMA_ERROR_LOG, prefix.c_str(), "Initalization of plugin failed. Failed to alloc execbo");
-                return XMA_ERROR;
+        try {
+            for (int32_t d = 0; d < count; d++) {
+                priv->kernel_execbos.emplace_back(XmaHwExecBO{});
+                XmaHwExecBO& dev_execbo = priv->kernel_execbos.back();
+                std::string cu_name = std::string(reinterpret_cast<char*>(priv->kernel_info->name));
```

Comment:
```suggestion
                std::string cu_name { static_cast<char*>(priv->kernel_info->name) };
```

Reply:
seeing compilation error if we use static_cast here.  submitted latest changes in the PR.